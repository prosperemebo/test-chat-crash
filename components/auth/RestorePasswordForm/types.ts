export interface RestorePasswordFormProps {
    mode: 'reset' | 'set'
    onResetPasswordPress: (email: string) => void
    onSetPasswordPress: (password: string) => void
    loading: boolean
}