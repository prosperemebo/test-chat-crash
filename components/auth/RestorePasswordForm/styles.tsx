import styled from 'styled-components/native'

export const FormWrapper = styled.View`
    width: 100%;
    margin-top: 40px;
`

export const InputsWrapper = styled.View`
    margin-bottom: 20px;
`

export const Title = styled.Text`
    color: ${props => props.theme.auth.forgotPasswordLabel};
    font-weight: 400;
    font-size: 13px;
    text-align: center;
    font-family: 'Poppins-Regular';
    margin-bottom: 10px;
`