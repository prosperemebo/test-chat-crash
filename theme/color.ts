export const WHITE = '#FFFFFF'
export const t10WHITE = 'rgba(255,255,255,0.1)'
export const t76WHITE = 'rgba(255,255,255,0.76)'

export const RED = '#f60004'
export const RED1 = '#D33E41'


export const BLACK = '#000000'
export const BLACK2 = '#050608'
export const nBLACK = '#15161A'
export const nBLACK10 = '#1F2025'
export const nBLACK30 = '#28292E'
export const nBLACK40 = '#050608'

export const nGRAY10 = '#6B6E79'
export const nGRAY20 = '#858995'
export const nGRAY25 = '#54575e'
export const nGRAY30 = '#2f3137'
export const nGRAY35 = '#52555D'
export const nGRAY40 = '#35363B'


export const tGRAY20 = '#85899599'
export const tGRAY30 = '#545458A6'
export const tGRAY40 = '#45464C'
export const GRAY40 = '#cfcfd0'
export const GRAY45 = '#c9c9cb'
export const GRAY50 = '#aaaaaa'
export const GRAY60 = '#52545D'
export const t40GRAY60 = '#52545D66'
export const GRAY70 = '#383B40'
export const GRAY90 = '#27292d'
export const GRAY100 = '#1f2023'
export const GRAY110 = '#151619'

export const GRAY_ONBOARD = '#8791a5'

export const BLUE = '#50b5e8'
export const BLUE2 = '#0A84FF'

export const PURPLE = '#9A1F99'

export const nBLUE = '#8D88A8'
export const nPURPLE = '#C47AF5'
export const nPURPLE10 = '#873FBB'
export const nPURPLE20 = '#873FBB'
export const nPURPLE40 = '#C882F8'

export const nORANGE10 = '#FF6705'
export const nORANGE20 = '#FF7905'
export const nORANGE40 = '#FFB700'

export const YELLOW2 = '#EBC789'
export const nYELLOW = '#F5D193'

export const nGRAYBACKGROUND2 = '#292A30'
export const nGRAYBACKGROUND3 = '#383940'
export const nGRAYBACKGROUND4 = '#989898'

export const gradient = {
  COLORS: [BLUE, PURPLE],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}

export const orangeGradient = {
  COLORS: [nORANGE10, nORANGE40],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}