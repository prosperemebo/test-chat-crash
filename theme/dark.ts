import {
  WHITE,
  BLACK,
  nGRAY10,
  GRAY_ONBOARD,
  nPURPLE,
  nBLUE,
  GRAY40,
  nGRAY20,
  RED1,
  tGRAY20,
  RED,
  nBLACK,
  nORANGE10,
  nORANGE40,
  nPURPLE10,
  GRAY70,
  tGRAY30,
  BLUE2,
  BLACK2,
  nGRAY25,
  nBLACK30,
  GRAY50,
  nBLACK10,
  t10WHITE,
  nGRAYBACKGROUND2,
  GRAY100,
  GRAY60,
  nGRAY40,
  nGRAY30,
  YELLOW2,
  nYELLOW,
  nGRAYBACKGROUND3,
  tGRAY40,
  nGRAY35,
  nBLACK40,
  t40GRAY60,
  nPURPLE20,
  nPURPLE40,
} from './color'

export const dark = {
  tint: nPURPLE,
  primaryTextColor: WHITE,
  primaryBg: nBLACK10,
  headerBg: GRAY100,
  headerBorderColor: GRAY60,
  secondaryTextColor: GRAY60,
  secondaryBgLight: nBLACK30,
  secondaryBg: nBLACK,
  serviceBg: GRAY100,
  inActiveBorderColor2: t10WHITE,
  backgroundColor2: nGRAYBACKGROUND2,
  backgroundColor3: nGRAYBACKGROUND3,
  backgroundColor4: nBLACK10,

  auth: {
    socialIconColor: WHITE,
    socialLoginButton: BLACK,
    dividerColor: nGRAY10,
    noticeText: WHITE,
    forgotPasswordLabel: WHITE,
  },

  onboarding: {
    headerTextColor: WHITE,
    headerStepsTextColor: GRAY_ONBOARD,
    progressBarColor: nPURPLE,
    progressBarBackgroundColor: nBLUE,
    selectedItemTextColor: WHITE,
    disabledItemTextColor: GRAY40,
    gameSectionOverlayColor: nPURPLE,
    gameEmptySectionColor: BLACK,
    gameEmptyIconColor: BLACK2,
  },

  common: {
    labelColor: WHITE,
    infoTextColor: nGRAY20,
    inputIconColor: nGRAY20,
    inputPlaceholderColor: nGRAY20,
    serviceTextColor: WHITE,
    inputTextColor: WHITE,
    inputTextError: RED1,
    inputBorderColor: tGRAY20,
    inputErrorBorderColor: RED,
    inputBackgroundColor: nBLACK,
    toastBackground: nBLACK,
    toastBorderColor: nPURPLE,
    toastTextColor: WHITE,
    buttonGradientOrangeStart: nORANGE10,
    buttonGradientOrangeEnd: nORANGE40,
    buttonGradientPurpleStart: nPURPLE40,
    buttonGradientPurpleEnd: nPURPLE20,
    buttonGradientGrayStart: nGRAY10,
    buttonGradientGrayEnd: nGRAY20,
    buttonDartBackgroundColor: BLACK,
    backButtonColor: nGRAY20,
    switchTextColor: WHITE,
    checkboxIconColor: nPURPLE,
    gradientButtonColorStart: nPURPLE10,
    gradientButtonColorEnd: nPURPLE,
    borderColor: GRAY70,
    genderPickerBorderColor: nGRAY25,
    genderPickerActiveTextColor: WHITE,
    genderPickerTextColor: GRAY50,
    checkboxTextColor: WHITE,
    iconButtonColor: WHITE,
    backgroundColor: nBLACK10,
  },

  settings: {
    socialIconBackgroundColor: nGRAY40,
    premiumFeatureTextColor: nPURPLE,
    buttonIconColor: nGRAY20,
    serviceItemBgColor: nGRAY30,
  },

  postOnboarding: {
    backgroundColor: nBLACK10,
    inputIconColor: nGRAY20,
    inputIconErrorColor: RED1,
    disclosureTextColor: nGRAY20,
    loadingIndicatorColor: BLACK,
  },

  profile: {
    backgroundColor: nBLACK10,
    improveBackgroundColor: YELLOW2,
    improveBannerBackgroundColor: nYELLOW,
    improveBannerTitleColor: nBLACK30,
    coverPictureCardRemoveButtonBg: tGRAY30,
    profilePicturesCountColor: tGRAY40,
  },

  chats: {
    iconColor: nGRAY20,
    inputColor: nBLACK40,
    placeholderColor: nGRAY35,
    appbarBorderColor: t40GRAY60,
    backgroundColor: nBLACK10,
  },

  nav: {
    bottomTabLabelBackgroundColor: nBLACK,
    bottomTabBackgroundColor: nBLACK,
  },

  other: {
    updateApp: {
      textBorderColor: tGRAY30,
      buttonTextColor: BLUE2,
      textColor: WHITE,
    },
  },
}
