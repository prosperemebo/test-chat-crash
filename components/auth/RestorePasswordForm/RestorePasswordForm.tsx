import { FormWrapper, Title, InputsWrapper } from './styles'
import i18n from '@/localization/i18n'
import { Button, Input } from '@/components/common'
import { useState } from 'react'
import AlertService from '@/services/Alert'
import { regex } from '@/utils/validation'
import { RestorePasswordFormProps } from './types'


const RestorePasswordForm = (props: RestorePasswordFormProps) => {
  const { mode, onSetPasswordPress, onResetPasswordPress, loading } = props

  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [passwordConfirm, setPasswordConfirm] = useState('')

  const buttonTitle = mode === 'reset' ? i18n.t('login:send') : i18n.t('login:restore')

  const isFieldNotValid = (field: string) => {
    let error = null
    switch (field) {
      case 'email':
        if (email.length < 1) error = i18n.t('errors:emptyEmailError')
        else if (!regex.email.test(email)) error = i18n.t('errors:invalidEmailError')
        break
      case 'password':
        if (password.length < 1) error = i18n.t('errors:emptyPasswordError')
        break
      case 'passwordConfirm':
        if (passwordConfirm.length < 1) error = i18n.t('errors:emptyPasswordError')
        if (passwordConfirm !== password) error = i18n.t('errors:passwordsMatchError')
        break
    }
    if (error) AlertService.showInfo(error)
    return error
  }

  const handleButtonPress = () => {
    if (mode === 'reset') {
      if (!!isFieldNotValid('email')) return
      onResetPasswordPress(email)
      return
    }

    if (!!isFieldNotValid('password')) return
    if (!!isFieldNotValid('passwordConfirm')) return

    onSetPasswordPress(password)
  }

  const renderContent = () => {
    if (mode === 'reset') {
      return <InputsWrapper>
        <Input type={'email'} value={email} placeholder={i18n.t('common:email')} onChangeText={setEmail} iconName={'envelope'} />
      </InputsWrapper>
    }

    return <InputsWrapper>
      <Input type={'password'} value={password} placeholder={i18n.t('login:newPassword')} onChangeText={setPassword} iconName={'key'} />
      <Input type={'password'} value={passwordConfirm} placeholder={i18n.t('login:confirmNewPassword')} onChangeText={setPasswordConfirm} iconName={'key'} />
    </InputsWrapper>
  }

  return <FormWrapper>
    <Title>{i18n.t('login:restorePassword')}</Title>
    {renderContent()}
    <Button title={buttonTitle}
      type={'dark'}
      onPress={handleButtonPress}
      loading={loading}
      disabled={loading}
    />
  </FormWrapper>
}

export default RestorePasswordForm