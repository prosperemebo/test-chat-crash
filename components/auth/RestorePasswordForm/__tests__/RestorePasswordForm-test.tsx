import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { describe, expect, test } from '@jest/globals'
import RestorePasswordForm from '../'
import { dark } from '@/theme'

describe('<RestorePasswordForm />', () => {

  test('RestorePasswordForm renders correctly', () => {
    const tree = render(
      <ThemeProvider theme={dark}>
        <RestorePasswordForm loading={true} mode={'reset'} onResetPasswordPress={() => {}} onSetPasswordPress={() => {}} />
        <RestorePasswordForm loading={false} mode={'reset'} onResetPasswordPress={() => {}} onSetPasswordPress={() => {}} />
        <RestorePasswordForm loading={false} mode={'set'} onSetPasswordPress={() => {}} onResetPasswordPress={() => {}} />
      </ThemeProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})