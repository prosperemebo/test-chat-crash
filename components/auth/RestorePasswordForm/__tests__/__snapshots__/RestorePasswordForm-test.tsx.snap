// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestorePasswordForm /> RestorePasswordForm renders correctly 1`] = `
[
  <View
    style={
      {
        "marginTop": 40,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Poppins-Regular",
          "fontSize": 13,
          "fontWeight": "400",
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      MOCKED_STRING:login:restorePassword
    </Text>
    <View
      style={
        {
          "marginBottom": 20,
        }
      }
    >
      <View
        showError={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#15161A",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderColor": "#85899599",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          isError={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#858995",
                "fontSize": 20,
                "marginRight": 10,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "400",
              },
            ]
          }
        >
          
        </Text>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="MOCKED_STRING:common:email"
          placeholderTextColor="#858995"
          style={
            {
              "color": "#FFFFFF",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "height": 32,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            }
          }
          textContentType="emailAddress"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#6B6E79",
          "borderBottomLeftRadius": 25,
          "borderBottomRightRadius": 25,
          "borderTopLeftRadius": 25,
          "borderTopRightRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 15,
          "opacity": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Poppins-Bold",
              "fontSize": 13,
              "marginRight": 5,
            }
          }
        >
          MOCKED_STRING:login:please_wait
        </Text>
        <ActivityIndicator
          color="#fff"
          size="small"
          style={
            {
              "alignSelf": "center",
            }
          }
        />
      </View>
    </View>
  </View>,
  <View
    style={
      {
        "marginTop": 40,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Poppins-Regular",
          "fontSize": 13,
          "fontWeight": "400",
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      MOCKED_STRING:login:restorePassword
    </Text>
    <View
      style={
        {
          "marginBottom": 20,
        }
      }
    >
      <View
        showError={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#15161A",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderColor": "#85899599",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          isError={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#858995",
                "fontSize": 20,
                "marginRight": 10,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "400",
              },
            ]
          }
        >
          
        </Text>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="MOCKED_STRING:common:email"
          placeholderTextColor="#858995"
          style={
            {
              "color": "#FFFFFF",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "height": 32,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            }
          }
          textContentType="emailAddress"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#6B6E79",
          "borderBottomLeftRadius": 25,
          "borderBottomRightRadius": 25,
          "borderTopLeftRadius": 25,
          "borderTopRightRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 15,
          "opacity": 1,
          "overflow": "hidden",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Poppins-Bold",
            "fontSize": 14,
            "textAlign": "center",
          }
        }
      >
        MOCKED_STRING:login:send
      </Text>
    </View>
  </View>,
  <View
    style={
      {
        "marginTop": 40,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Poppins-Regular",
          "fontSize": 13,
          "fontWeight": "400",
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      MOCKED_STRING:login:restorePassword
    </Text>
    <View
      style={
        {
          "marginBottom": 20,
        }
      }
    >
      <View
        showError={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#15161A",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderColor": "#85899599",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          isError={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#858995",
                "fontSize": 20,
                "marginRight": 10,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="MOCKED_STRING:login:newPassword"
          placeholderTextColor="#858995"
          secureTextEntry={true}
          style={
            {
              "color": "#FFFFFF",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "height": 32,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            }
          }
          textContentType="password"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            isError={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                {
                  "color": "#858995",
                  "fontSize": 20,
                  "marginRight": 10,
                },
                {
                  "fontFamily": undefined,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "400",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        showError={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#15161A",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderColor": "#85899599",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          isError={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#858995",
                "fontSize": 20,
                "marginRight": 10,
              },
              {
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="MOCKED_STRING:login:confirmNewPassword"
          placeholderTextColor="#858995"
          secureTextEntry={true}
          style={
            {
              "color": "#FFFFFF",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "height": 32,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            }
          }
          textContentType="password"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            isError={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 12,
                },
                {
                  "color": "#858995",
                  "fontSize": 20,
                  "marginRight": 10,
                },
                {
                  "fontFamily": undefined,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "400",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#6B6E79",
          "borderBottomLeftRadius": 25,
          "borderBottomRightRadius": 25,
          "borderTopLeftRadius": 25,
          "borderTopRightRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 15,
          "opacity": 1,
          "overflow": "hidden",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Poppins-Bold",
            "fontSize": 14,
            "textAlign": "center",
          }
        }
      >
        MOCKED_STRING:login:restore
      </Text>
    </View>
  </View>,
]
`;
